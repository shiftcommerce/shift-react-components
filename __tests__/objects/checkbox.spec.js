// Libraries
import React from 'react'

// Objects
import Checkbox from '../../src/objects/checkbox'

test('renders correctly', () => {
  // Arrange
  let fieldOption = {
    label: 'Sign up for Weekly Newsletters (optional)',
    name: 'newsletterOptIn',
    value: true
  }

  // Act
  const wrapper = mount(
    <Checkbox
      label={fieldOption.label}
      name={fieldOption.name}
      checked={!!fieldOption.value}
    />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('label')).toIncludeText(fieldOption.label)
  expect(wrapper.find('input')).toHaveProp('name', fieldOption.name)
  expect(wrapper.find('input')).toHaveProp('checked', fieldOption.value)
})
