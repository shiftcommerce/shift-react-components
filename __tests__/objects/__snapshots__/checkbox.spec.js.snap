// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<_class
  checked={true}
  label="Sign up for Weekly Newsletters (optional)"
  name="newsletterOptIn"
>
  <Checkbox
    checked={true}
    label="Sign up for Weekly Newsletters (optional)"
    name="newsletterOptIn"
    renderValidationMessage={[Function]}
  >
    <div
      className="o-form__input-group"
    >
      <label
        className="o-form__label"
        htmlFor="newsletterOptIn"
      >
        <input
          checked={true}
          className="o-form__checkbox"
          id="newsletterOptIn"
          name="newsletterOptIn"
          onBlur={[Function]}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="o-form__checkbox-caption"
        >
          Sign up for Weekly Newsletters (optional)
           
        </span>
      </label>
    </div>
  </Checkbox>
</_class>
`;
